sdk/tools/cabman/CCFDATAStorageUnix.cxx

```diff
@@ -0,0 +1,169 @@
+/*
+* COPYRIGHT:   See COPYING in the top level directory
+* PROJECT:     ReactOS cabinet manager
+* FILE         tools/cabman/CCFDATAStorageWindows.cxx
+* PURPOSE:     CCFDATAStorage class implementation for Windows
+* PROGRAMMERS:
+* NOTES:
+* REVISIONS:
+*/
+#if !defined(_WIN32)
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <dirent.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+
+#include "cabinet.h"
+#include "raw.h"
+#include "mszip.h"
+
+#if !defined(CAB_READ_ONLY)
+
+CCFDATAStorage::CCFDATAStorage()
+/*
+* FUNCTION: Default constructor
+*/
+{
+    FileCreated = false;
+}
+
+CCFDATAStorage::~CCFDATAStorage()
+/*
+* FUNCTION: Default destructor
+*/
+{
+    ASSERT(!FileCreated);
+}
+
+ULONG CCFDATAStorage::Create()
+/*
+* FUNCTION: Creates the file
+* ARGUMENTS:
+*     FileName = Pointer to name of file
+* RETURNS:
+*     Status of operation
+*/
+{
+    /*if (tmpnam(FullName) == NULL)*/
+    if ((FileHandle = tmpfile()) == NULL)
+        return CAB_STATUS_CANNOT_CREATE;
+    /*
+    FileHandle = fopen(FullName, "w+b");
+    if (FileHandle == NULL) {
+        DPRINT(MID_TRACE, ("ERROR '%i'.\n", errno));
+        return CAB_STATUS_CANNOT_CREATE;
+    }
+    */
+
+    FileCreated = true;
+
+    return CAB_STATUS_SUCCESS;
+}
+
+ULONG CCFDATAStorage::Destroy()
+/*
+* FUNCTION: Destroys the file
+* RETURNS:
+*     Status of operation
+*/
+{
+    ASSERT(FileCreated);
+
+    CloseFile(FileHandle);
+
+    FileCreated = false;
+
+    return CAB_STATUS_SUCCESS;
+}
+
+
+ULONG CCFDATAStorage::Truncate()
+/*
+* FUNCTION: Truncate the scratch file to zero bytes
+* RETURNS:
+*     Status of operation
+*/
+{
+    fclose(FileHandle);
+    FileHandle = tmpfile();
```

Comment:
Why such a complex implementation?
Why not using ftruncate(fileno(FileHandle), 0); ?
And likely fseek(FileHandle, 0, SEEK_SET); beforehand.

Reply:
Agree that this is not an optimal implementation, but current task is only to remove _WIN32, so I just left the implementation itself untouched. 
Refactoring could be the next step.

Reply:
OK for me then.